projects/views.py

```diff
@@ -208,28 +233,24 @@ def create_environment(request, project_slug):
 @login_required
 @permission_required_or_403("can_view_project", (Project, "slug", "project_slug"))
 def delete_environment(request, project_slug):
-    if request.method == "POST":
-        project = Project.objects.get(slug=project_slug)
-        pk = request.POST.get("environment_pk")
-        # TODO: Check that the user has permission to delete this environment.
-        environment = Environment.objects.get(pk=pk, project=project)
-
-        can_environment_be_deleted = True
-        for app_orm in APP_REGISTRY.iter_orm_models():
-            if hasattr(app_orm, "environment"):
-                queryset = app_orm.objects.filter(environment=environment)
-                if queryset:
-                    messages.error(
-                        request,
-                        "Environment cannot be deleted because it is currently used by at least one app \
-                            (can also be a deleted app).",
-                    )
-                    can_environment_be_deleted = False
-                    break
+    project = Project.objects.get(slug=project_slug)
+    if not request.user.is_superuser:
+        return HttpResponseForbidden()
+    else:
+        if request.method == "POST":
+            pk = request.POST.get("environment_pk")
+            environment = Environment.objects.get(pk=pk, project=project)
 
-        if can_environment_be_deleted:
-            environment.delete()
+            can_environment_be_deleted = can_model_instance_be_deleted("environment", pk)
 
+            if can_environment_be_deleted:
+                environment.delete()
+            else:
+                messages.error(
+                    request,
+                    "Environment cannot be deleted because it is currently used by at least one app \
+                        (can also be a deleted app).",
```

Comment:
What to do if it's a deleted app?

Reply:
I think then we are not allowed to delete that flavor. If there is a use case where it is needed we can implement something else later I think. It may have repercussions further down, not sure.